src/main/java/de/mediathekview/mserver/crawler/wdr/parser/WdrFilmDeserializer.java

```diff
@@ -0,0 +1,308 @@
+
+        film.setWebsite(new URL(aUrlDto.getUrl()));
+        if (aDescription.isPresent()) {
+          film.setBeschreibung(aDescription.get());
+        }
+
+        WdrVideoInfoDto videoInfo = aVideoInfo.get();
+        if (StringUtils.isNotBlank(videoInfo.getSubtitleUrl())) {
+          film.addSubtitle(new URL(videoInfo.getSubtitleUrl()));
+        }
+
+        addUrls(film, videoInfo);
+        film.setGeoLocations(CrawlerTool.getGeoLocations(sender, videoInfo.getDefaultVideoUrl()));
+
+        return Optional.of(film);
+
+      } else {
+        LOG.error("WdrFilmDeserializer: no title or video found for url " + aUrlDto.getUrl());
+      }
+    } catch (MalformedURLException ex) {
+      LOG.fatal("WdrFilmDeserializer: url can't be parsed.", ex);
+    }
+
+    return Optional.empty();
+  }
+
+  private void addUrls(final Film aFilm, final WdrVideoInfoDto aVideoInfo)
+      throws MalformedURLException {
+
+    Map<Resolution, String> videoUrls = aVideoInfo.getVideoUrls();
+    for (final Map.Entry<Resolution, String> qualitiesEntry : videoUrls.entrySet()) {
+      aFilm.addUrl(qualitiesEntry.getKey(), CrawlerTool.stringToFilmUrl(qualitiesEntry.getValue()));
+    }
+
+    videoUrls = aVideoInfo.getAudioDescriptionUrls();
+    for (final Map.Entry<Resolution, String> qualitiesEntry : videoUrls.entrySet()) {
+      aFilm.addAudioDescription(
+          qualitiesEntry.getKey(), CrawlerTool.stringToFilmUrl(qualitiesEntry.getValue()));
+    }
+
+    videoUrls = aVideoInfo.getSignLanguageUrls();
+    for (final Map.Entry<Resolution, String> qualitiesEntry : videoUrls.entrySet()) {
+      aFilm.addSignLanguage(
+          qualitiesEntry.getKey(), CrawlerTool.stringToFilmUrl(qualitiesEntry.getValue()));
+    }
+  }
+
+  private Optional<WdrVideoInfoDto> parseUrls(final Document aDocument) {
+    final Optional<CrawlerUrlDTO> videoUrlDto = parseVideoLink(aDocument);
+    if (!videoUrlDto.isPresent()) {
+      return Optional.empty();
+    }
+    Optional<String> javaScriptContent = readContent(videoUrlDto.get().getUrl());
+    Optional<String> embeddedJson = extractJsonFromJavaScript(javaScriptContent);
+    if (!embeddedJson.isPresent()) {
+      return Optional.empty();
+    }
+
+    Optional<WdrMediaDto> mediaDto =
+        gson.fromJson(embeddedJson.get(), OPTIONAL_WDRMEDIADTO_TYPE_TOKEN);
+    if (!mediaDto.isPresent()) {
+      return Optional.empty();
+    }
+
+    WdrMediaDto dto = mediaDto.get();
+    if (dto.getUrl().endsWith(".m3u8")) {
+      return parseM3U8(dto);
+    } else if (dto.getUrl().endsWith(".mp4")) {
+      return parseMP4Url(dto);
+    }
+
+    return Optional.empty();
+  }
+
+  private Optional<WdrVideoInfoDto> parseM3U8(final WdrMediaDto aMediaDto) {
+    Map<Resolution, String> urlMap = parseM3U8Url(aMediaDto.getUrl());
+    if (!urlMap.isEmpty()) {
+      WdrVideoInfoDto videoInfoDto = new WdrVideoInfoDto();
+      urlMap.forEach((key, value) -> videoInfoDto.putVideo(key, value));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)